Bricks/net/tcp/impl/posix.h

```diff
@@ -483,18 +526,43 @@ inline Connection ClientSocket(const std::string& host, T port_or_serv) {
       // for (struct addrinfo* p = servinfo; p != NULL; p = p->ai_next) {
       //   p->ai_addr;
       // }
+      struct sockaddr_in* p_addr_in = reinterpret_cast<struct sockaddr_in*>(p_addr);
+      remote_ip_and_port.ip = InetAddrToString(&(p_addr_in->sin_addr));
+      remote_ip_and_port.port = htons(p_addr_in->sin_port);
+
       BRICKS_NET_LOG("S%05d connect() ...\n", static_cast<SOCKET>(socket));
       const int retval2 = ::connect(socket, p_addr, sizeof(*p_addr));
       if (retval2) {
         CURRENT_THROW(SocketConnectException());  // LCOV_EXCL_LINE -- Not covered by the unit tests.
       }
+
+      sockaddr_in addr_client;
+      auto addr_client_length = sizeof(sockaddr_in);
+#ifndef CURRENT_WINDOWS
+      if (::getsockname(socket,
+                        reinterpret_cast<struct sockaddr*>(&addr_client),
+                        reinterpret_cast<socklen_t*>(&addr_client_length)) == -1) {
```

Comment:
Nit here and above: If instead of `auto`, the variable cast into `socklen_t` is declared with the type specific to the platform, then the code, not data, part -- the `if (::getsockname(...) != 0)` --- would be identical on Windows and Linux.

I'll be able to test on Windows in about 4 hours.